frontend/packages/console-dynamic-plugin-sdk/src/coderefs/coderef-resolver.ts

```diff
@@ -99,15 +123,15 @@ export const resolveEncodedCodeRefs = (
   errorCallback: VoidFunction,
 ): Extension[] =>
   _.cloneDeep(extensions).map((e) => {
-    const refs = filterEncodedCodeRefProperties(e.properties);
-
-    Object.entries(refs).forEach(([propName, ref]) => {
+    const paths = getCodeRefPaths(e.properties, isEncodedCodeRef);
```

Comment:
I'd suggest the following approach:

- adapt `filterEncodedCodeRefProperties` and `filterExecutableCodeRefProperties` to deep-pick any code refs, for example:

```ts
filterEncodedCodeRefProperties({
  foo: true,
  bar: { "$codeRef": "a.b" },
  qux: {
    baz: { "$codeRef": "x.y" },
    test: 1
  }
})
// =>
{
  bar: { "$codeRef": "a.b" },
  qux: {
    baz: { "$codeRef": "x.y" }
  }
}
```

Code ref properties are ones whose value passes the given test (`isEncodedCodeRef` or `isExecutableCodeRef`).

- adapt `resolveEncodedCodeRefs` to deep-map the result of `filterEncodedCodeRefProperties` and merge it with extension's properties

- adapt `resolveCodeRefProperties` to deep-map the result of `filterExecutableCodeRefProperties` and return it

- adapt `collectCodeRefData` in extension validator

- `getDynamicExtensions` in active-plugins module uses `JSON.stringify` so this code shouldn't need any changes

And of course, adapt all affected unit tests :smiley: this is a lot of work and should be done carefully in a separate PR.